### PR TITLE
This update enables the "new" pjsip SIP module support for asterisk.

### PR DESCRIPTION
**Pull request by olbohlen**
The old "chan_sip" is no longer recommended by asterisk, "pjsip" is the
preferred variant now.

Unfortunately we disabled building pjsip support, because of some compiler
errors. This update includes a patch
(patches/06-my-sockaddr_in-to-enable-pjsip.patch) that works around the issue
by providing an alternative name for a conflicting definition.

### CODE REVIEW

**Comment by olbohlen** [comment]
Tested the code change here:

> root@testrun:~# su - asterisk -c '/usr/sbin/asterisk -rvvvvv'
> Asterisk 15.1.4, Copyright (C) 1999 - 2016, Digium, Inc. and others.
> Created by Mark Spencer <markster@digium.com>
> Asterisk comes with ABSOLUTELY NO WARRANTY; type 'core show warranty' for details.
> This is free software, with components licensed under the GNU General Public
> License version 2 and other licenses; you are welcome to redistribute it under
> certain conditions. Type 'core show license' for details.
> =========================================================================
> Connected to Asterisk 15.1.4 currently running on testrun (pid = 6578)
> testrun*CLI> pjsip 
> dump     export   list     qualify  show     
> testrun*CLI> pjsip show 
> aor                    aors                   auth                   
> auths                  contact                contacts               
> endpoint               endpoints              identifiers            
> identifies             identify               scheduled_tasks        
> settings               subscription           subscriptions          
> transport              transports             unidentified_requests  
> version                
> testrun*CLI> pjsip show version
> PJPROJECT version currently running against: 2.6
> testrun*CLI> 
>

File: components/network/asterisk/Makefile [hunk 1]
```diff
@@ -11,6 +11,7 @@
 #
 # Copyright 2016 Andrey Sokolov. All rights reserved.
 # Copyright 2017 Alexander Pyhalov
+# Copyright 2021 Olaf Bohlen
 #
```
Comment (AndWac) [inline components/network/asterisk/Makefile:16]: BUILD_BITS= 32
is missing here

File: components/network/asterisk/Makefile [hunk 2]
```diff
@@ -26,7 +27,9 @@ COMPONENT_ARCHIVE_URL=	http://downloads.asterisk.org/pub/telephony/asterisk/old-
 COMPONENT_LICENSE=	GPLv2
 COMPONENT_LICENSE_FILE=	$(COMPONENT_NAME).license
 
-include $(WS_MAKE_RULES)/common.mk
+include $(WS_TOP)/make-rules/prep.mk
+include $(WS_TOP)/make-rules/configure.mk
+include $(WS_TOP)/make-rules/ips.mk
```
Comment (AndWac) [inline components/network/asterisk/Makefile:36]: You changed from the new style of imports to the old one. Please revert that and add TEST_TARGET like so:
TEST_TARGET= $(NO_TESTS)
include $(WS_MAKE_RULES)/common.mk

File: components/network/asterisk/Makefile [hunk 3]
```diff
@@ -47,12 +50,17 @@ CONFIGURE_SBINDIR=	/usr/sbin
...
 
 COMPONENT_INSTALL_TARGETS += samples
 
+build:		$(BUILD_32)
+
+install:	$(INSTALL_32)
+
+test:		$(NO_TESTS)
+
 # Build dependencies
```
Comment (AndWac) [inline components/network/asterisk/Makefile]: Please remove the explicit targets. They are superfluous with BUILD_BITS= 32, TEST_TARGET=$(NO_TESTS) and include $(WS_MAKE_RULES)/common.mk

File: components/network/asterisk/Makefile (hunk 1 above)
Comment (AndWac) [inline components/network/asterisk/Makefile:16]: When you make changes to a package while not upgrading the COMPONENT_VERSION you'll need to set or increment its COMPONENT_REVISION! This is missing here.

**Review by AndWac — APPROVED** [review]

**Pull request merged**